cluster/Vagrantfile

```diff
@@ -1,116 +1,137 @@
-# -*- mode: ruby -*-
+  # -*- mode: ruby -*-
 # vi: set ft=ruby :
 require 'rubygems'
-require 'json'
 require 'fileutils'
+require 'yaml'
 
 token = 'd900e1.8a392798f13b33a4'
 
 # method to create an etc_hosts file based on the cluster info
-def create_etc_hosts(cluster)
-  master_ip = '192.168.2.10'
+def create_etc_hosts(num_nodes, base_ip, start)
+  master_ip = base_ip + start.to_s
   hosts = "127.0.0.1   localhost\n"
-  cluster.each do |role, member_list|
-    hosts = member_list.inject(hosts) { |acc, elem| acc << "#{elem['contiv_control_ip']}   #{elem['name']}\n" }
-    if role == 'master' && member_list[0]
-      hosts << "#{member_list[0]['contiv_control_ip']}   netmaster\n"
-      master_ip = member_list[0]['contiv_control_ip']
-    end
-  end
+  hosts << "#{master_ip}   netmaster\n"
+  hosts = (0..num_nodes).inject(hosts) { |acc, elem| acc << base_ip + "#{elem + start} contiv-node#{elem + 1} \n" }
```

Comment:
One thing to be careful of with #inject/#reduce, the result of evaluating the block becomes `acc` on the next iteration, so in a lot of cases you'll need to explicitly put `acc` as the last thing in the block after adding to/modifying it.

In this case, you're fine because `String#<<` returns the complete string.  Just something to be aware of if you start getting really funky in #inject/#reduce